cells/lib/ops/mkStandardOCI.nix

```diff
@@ -68,20 +68,82 @@ in
       then debugOperable
       else operable;
 
-    setupLinks = cell.ops.mkSetup "links" [] ''
-      mkdir -p $out/bin
-      ${runtimeEntryLink}
-      ${debugEntryLink}
-      ${livenessLink}
-      ${readinessLink}
+    setupLinks =
+      cell.ops.mkSetup "links" [
+        {
+          regex = "/bin";
+          mode = "0555";
+        }
+      ] ''
+        mkdir -p $out/bin
+        ${runtimeEntryLink}
+        ${debugEntryLink}
+        ${livenessLink}
+        ${readinessLink}
+      '';
+
+    setupCerts = cell.ops.mkSetup "ssl" [] ''
+      mkdir -p $out/etc
+      cp -r ${nixpkgs.cacert}/* $out
+    '';
+
+    setupEtcHack = cell.ops.mkSetup "etc" [] ''
+      mkdir -p $out/etc
+      cat << REASON > $out/etc/.ensure-etc-is-no-symlink
+        # docker creates some minimal files in /etc at runtime
+        # that would otherwise shadow a top-level symlink:
+        #   bash-5.2$ ls /etc
+        #   hostname  hosts  mtab  resolv.conf
+        # therefore, we make trick nixpks.buildenv into not
+        # generating symlinks at the level of /etc but at least
+        # one level deeper by artifie of this overlapping element
+        # that buildenv will detect and push the linking one level down
+      REASON
+    '';
+
+    setupSparseUser = cell.ops.mkUser {
+      user = "nobody";
+      group = "nogroup";
+      inherit uid gid;
+    };
+
+    tmp = nixpkgs.runCommand "tmp" {} ''
+      mkdir -p $out/tmp
     '';
   in
     cell.ops.mkOCI (
       {
-        inherit name uid gid labels options perms config meta;
+        inherit name uid gid labels meta;
         entrypoint = operable';
-        setup = [setupLinks] ++ setup;
+        setup = [setupLinks setupCerts setupEtcHack] ++ setup;
```

Comment:
Isn't this forcing all images to have certs?

Reply:
Indeed, one reasoning may be that `mkStandardOCI` comes with the `curl` certificate authorities.

That's not what high assurance environments (e.g. medicine, military, transport, etc) would want, but there's `mkOCI`, still.

Reply:
I am not sure how I feel about it but I am happy that it resolves my issue

Reply:
It's a cutoff criteria that we need to apply here, and it's indeed not trivial to strike a good balanace.